ergo/distributions/point_density.py

```diff
@@ -0,0 +1,199 @@
+        return np.where(normed_x < 0, 0, np.where(normed_x > 1, 1, c))
+
+    def ppf(self, q):
+        return self.scale.denormalize_point(
+            self.normed_xs[np.argmax(self.cumulative_normed_ps >= q)]
+        )
+
+    def sample(self):
+        raise NotImplementedError
+
+    # Scaled
+
+    def normalize(self):
+        return PointDensity(
+            self.normed_xs, self.normed_densities, scale=Scale(0, 1), normalized=True
+        )
+
+    def denormalize(self, scale: Scale):
+        return PointDensity(
+            self.normed_xs, self.normed_densities, scale=scale, normalized=True
+        )
+
+    # Structured
+
+    @classmethod
+    def structure(cls, params):
+        class_params, numeric_params = params
+        density_class, scale_classes = class_params
+        density_numeric, scale_numeric = numeric_params
+        return density_class(
+            xs=density_numeric[0],
+            densities=density_numeric[1],
+            scale=scale_classes[0].structure((scale_classes, scale_numeric)),
+            normalized=True,
+            traceable=True,
+            cumulative_normed_ps=density_numeric[2],
+        )
+
+    def destructure(self):
+        scale_classes, scale_numeric = self.scale.destructure()
+        class_params = (self.__class__, scale_classes)
+        self_numeric = self.normed_xs, self.normed_densities, self.cumulative_normed_ps
+        numeric_params = (self_numeric, scale_numeric)
+        return (class_params, numeric_params)
+
+    # Optimizable
+
+    @classmethod
+    def from_conditions(cls, *args, fixed_params=None, scale=None, **kwargs):
+        if scale is None:
+            # TODO: Should we do this?
+            scale = Scale(0, 1)
+        if fixed_params is None:
+            fixed_params = {
+                "xs": np.linspace(
+                    scale.low, scale.high, constants.point_density_default_num_points
+                )
+            }
+        return super(PointDensity, cls).from_conditions(
+            *args, fixed_params=fixed_params, scale=scale, **kwargs
+        )
+
+    @classmethod
+    def from_params(cls, fixed_params, opt_params, scale=None, traceable=True):
+        # TODO: traceable is always True here
+        if scale is None:
+            # TODO: Should we do this?
+            scale = Scale(0, 1)
+        xs = fixed_params["xs"]
+        ps = np.abs(opt_params)
+        Z = np.sum(cls.normed_bin_probs(xs, ps))
+        densities = ps / Z
+        # print(np.sum(cls.normed_bin_probs(xs, densities)))
+        return cls(
+            xs=xs, densities=densities, scale=scale, normalized=True, traceable=True
+        )
+
+    @staticmethod
+    def initialize_optimizable_params(fixed_params):
+        num_points = fixed_params["xs"].size
+        return onp.full(num_points, 1.0)
+
+    @classmethod
+    def normalize_fixed_params(self, fixed_params, scale):
+        return {"xs": scale.normalize_points(fixed_params["xs"])}
```

Comment:
`normalize_points` isn't vectorized, so this could lead to some pretty slow Jax compilations or slow compiled computations.